src/ResponseHeaders.php

```diff
@@ -0,0 +1,280 @@
+            header($name . ': ' . implode(', ', (array)$values));
+        }
+    }
+    
+    /**
+     * For the current object simple chek  if current object 
+     * can be used for set or modify or remove headers
+     *
+     * @throws \InvalidArgumentException
+     */
+    protected function assertStale()
+    {
+        if ($this->isStale !== null) {
+            throw new \InvalidArgumentException("Can not change stale object");
+        }
+    }
+
+    /**
+     * Change current object to stale, move all headers to the variable
+     */
+    protected function setClassStale()
+    {
+        $this->isStale = true;
+        
+        $list = headers_list();
+        foreach ($list as $header) {
+            list($key, $value) = explode(': ', $header);
+            $this->headers[strtolower($key)] = ['name' => $key, 'values' => explode(', ', $value)];
+        }
+    }
+
+    /**
+     * Retrieves all message header values.
+     *
+     * The keys represent the header name as it will be sent over the wire, and
+     * each value is an array of strings associated with the header.
+     *
+     * // Represent the headers as a string
+     * foreach ($message->getHeaders() as $name => $values) {
+     * echo $name . ': ' . implode(', ', $values);
+     * }
+     *
+     * // Emit headers iteratively:
+     * foreach ($message->getHeaders() as $name => $values) {
+     * foreach ($values as $value) {
+     * header(sprintf('%s: %s', $name, $value), false);
+     * }
+     * }
+     *
+     * @return string[][] Returns an associative array of the message's headers.
+     *         Each key is a header name, and each value is an array of strings for
+     *         that header.
+     */
+    public function getHeaders()
+    {
+        if ($this->isStale) {
+            return parent::getHeaders();
+        }
+        
+        $headers = [];
+        
+        $list = headers_list();
+        foreach ($list as $header) {
+            list($key, $value) = explode(': ', $header);
+            $headers[$key] = explode(', ', $value);
+        }
+        
+        return $headers;
+    }
+
+    /**
+     * Checks if a header exists by the given case-insensitive name.
+     *
+     * @param string $name
+     *            Case-insensitive header field name.
+     * @return bool Returns true if any header names match the given header
+     *         name using a case-insensitive string comparison. Returns false if
+     *         no matching header name is found in the message.
+     */
+    public function hasHeader($name)
+    {
+        $this->assertHeaderName($name);
+        
+        if ($this->isStale) {
+            return parent::hasHeader($name);
+        }
+        
+        $headers = headers_list();
+        foreach ($headers as $header) {
+            list($key) = explode(': ', $header);
+            if (strtolower($key) == strtolower($name)) {
+                return true;
+            }
+        }
```

Comment:
This code is duplicated (twice). Instead add `protected function getRawHeader()`, that returns the header value.